hnn_core/network.py

```diff
@@ -947,29 +947,47 @@ def _all_to_all_connect(self, src_cell, target_cell,
             all target_type cells.
         """
         src_gids = self.gid_ranges[_long_name(src_cell)]
-        target_gids = self.gid_ranges[_long_name(target_cell)]
+        target_range = self.gid_ranges[_long_name(target_cell)]
 
         src_start = src_gids[0]  # Necessary for unique feeds
-        for target_gid in target_gids:
-            if unique:
-                src_gids = [target_gid + src_start]
-            for src_gid in src_gids:
-                if not allow_autapses and src_gid == target_gid:
-                    continue
-                self.add_connection(
-                    src_gid, target_gid, loc, receptor,
-                    nc_dict['A_weight'], nc_dict['A_delay'], nc_dict['lamtha'])
 
-    def add_connection(self, src_gid, target_gid, loc, receptor,
+        if unique:
+            src_gids = np.array(target_range) + src_start
+            src_gids = src_gids.astype(int).tolist()
```

Comment:
why convert to float array and then back to list?

```suggestion
            src_gids = [src_gid + src_start for src_gid in target_range]
```